xmake/rules/protobuf/proto.lua

```diff
@@ -185,21 +218,32 @@ function build_batchjobs(target, batchjobs, sourcebatch, opt, sourcekind)
     opt = opt or {}
     local moduledeps, moduledeps_files = module_parser.load(target, sourcebatch, table.join(opt, {proto_rootdir = proto_rootdir}))
 
-    -- generate jobs
     local sourcefiles_total = #sourcebatch.sourcefiles
+    -- generate jobs
     for i = 1, sourcefiles_total do
         local sourcefile = sourcebatch.sourcefiles[i]
         local moduleinfo = moduledeps_files[sourcefile] or {}
-
         -- make build job
         moduleinfo.job = batchjobs:newjob(sourcefile, function (index, total)
             local batchcmds_ = batchcmds.new({target = target})
-            buildcmd(target, batchcmds_, sourcefile, {progress = (index * 100) / total}, sourcekind)
+            build_proto_cmd(target, batchcmds_, sourcefile, {progress = (index * 100) / total}, sourcekind)
             batchcmds_:runcmds({changed = target:is_rebuilt(), dryrun = option.get("dry-run")})
         end)
     end
-
-    -- build batchjobs
     module_parser.build_batchjobs(moduledeps, batchjobs, opt.rootjob)
 end
 
+function build_compiled_sourcefiles_proto(target, batchjobs, sourcebatch, opt, sourcekind)
+    opt = opt or {}
+    local sourcefiles_total = #sourcebatch.sourcefiles
+    -- generate jobs
+    for i = 1, sourcefiles_total do
+        local sourcefile = sourcebatch.sourcefiles[i]
+        -- make build job
+        batchjobs:newjob(sourcefile, function (index, total)
```

Comment:
it only create a job, you did not add job to build batch. so it does nothing now. Did you test it?

try remove `build` and rebuild it.

Reply:
you can call private.action.build.object to pass correct sourcebatch. maybe we need create a new sourcebatch with cpp files

https://github.com/xmake-io/xmake/blob/7b1a926435cdd81c9b09a2eedfa43fe7c00c3302/xmake/rules/c%2B%2B/xmake.lua#L24

Reply:
> it only create a job, you did not add job to build batch. so it does nothing now. Did you test it?
> 
> try remove `build` and rebuild it.

If the build batch does nothing, then maybe it isn't needed since I tested on the sample proto project, sample protobuf_grpc_cpp_plugin project and the cartographer repository; and the generated cpp files still compiled